plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/ScaleIOStorageAdaptor.java

```diff
@@ -268,12 +269,20 @@ public KVMPhysicalDisk copyPhysicalDisk(KVMPhysicalDisk disk, String name, KVMSt
             srcFile = new QemuImgFile(disk.getPath(), disk.getFormat());
             destFile = new QemuImgFile(destDisk.getPath(), destDisk.getFormat());
 
-            LOGGER.debug("Starting copy from source image " + srcFile.getFileName() + " to PowerFlex volume: " + destDisk.getPath());
+            LOGGER.debug("Starting copy from source disk image " + srcFile.getFileName() + " to PowerFlex volume: " + destDisk.getPath());
             qemu.convert(srcFile, destFile);
-            LOGGER.debug("Succesfully converted source image " + srcFile.getFileName() + " to PowerFlex volume: " + destDisk.getPath());
+            LOGGER.debug("Succesfully converted source disk image " + srcFile.getFileName() + " to PowerFlex volume: " + destDisk.getPath());
         }  catch (QemuImgException | LibvirtException e) {
-            LOGGER.error("Failed to convert from " + srcFile.getFileName() + " to " + destFile.getFileName() + " the error was: " + e.getMessage(), e);
-            destDisk = null;
+            try {
+                Map<String, String> srcInfo = qemu.info(srcFile);
+                LOGGER.debug("Source disk info: " + Arrays.asList(srcInfo));
+            } catch (Exception ignored) {
+                LOGGER.warn("Unable to get info from source disk: " + disk.getName());
+            }
+
+            String errMsg = "Unable to convert/copy from " + disk.getName() + " to " + name + ((Strings.isNullOrEmpty(e.getMessage())) ? "" : ", due to: " + e.getMessage());
+            LOGGER.error(errMsg, e);
+            throw new CloudRuntimeException(errMsg);
```

Comment:
how about a method `handleVirtualisationException(...)`, and
```suggestion
            try {
                Map<String, String> srcInfo = qemu.info(srcFile);
                LOGGER.debug("Source disk info: " + Arrays.asList(srcInfo));
            } catch (Exception ignored) {
                LOGGER.warn("Unable to get info from source disk: " + disk.getName());
            }

            String errMsg = String.format("Unable to convert/copy from %s to %s, due to: %s", disk.getName(), name, ((Strings.isNullOrEmpty(e.getMessage())) ? "an unknown error" : e.getMessage());
            LOGGER.error(errMsg, e);
            throw new CloudRuntimeException(errMsg);
```